types/types.go

```diff
@@ -292,8 +292,19 @@ const (
 	replicaManagerPrefix  = instanceManagerPrefix + "r-"
 )
 
-func GenerateEngineNameForVolume(vName string) string {
-	return vName + engineSuffix + "-" + util.RandomID()
+func GenerateEngineNameForVolume(vName, currentEngineName string) string {
+	if currentEngineName == "" {
+		return vName + engineSuffix + "-" + "0"
+	}
+	parts := strings.Split(currentEngineName, "-")
+	if len(parts) < 3 {
+		return vName + engineSuffix + "-" + "1"
+	}
```

Comment:
nit: have a unit test for this function.

Reply:
Added. Thank you